### PR TITLE
Added doc explaining how to create notes and occurrences, along with a

### PR DESCRIPTION
**Pull request by avh121**
'guides' section.

### CODE REVIEW

File: _docs/guides/create-notes-occurrences.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
+---
+title: How to create notes and occurrences
+overview: Describes how to create notes and occurrences once you have a running Grafeas server.
+
+order: 15
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md:5]: what is this for?

File: _docs/guides/create-notes-occurrences.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+You can add all of the following methods to your
+[GrafeasApiExample](https://github.com/nhayes/client-java/#getting-started)
+class, or the equivalent class that you're using to talk to the client library.
+
+### Create a note
+
+Add a note to the `provider_example` project as follows:
+
+```java
+/**
```
Comment (vtsao) [inline _docs/guides/create-notes-occurrences.md]: nit: alignment is off here

File: _docs/guides/create-notes-occurrences.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+title: How to create notes and occurrences
+overview: Describes how to create notes and occurrences once you have a running Grafeas server.
+
+order: 15
+
+layout: docs
+type: markdown
+---
+
+Before you can create [notes](/_docs/concepts/what-is-grafeas/overview.md) or
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: I would recommend adding a "Before you begin" section:

## Before you begin

+ Set up a Grafeas server running on port 8080 using the instructions in [Running Grafeas](https://github.com/grafeas/grafeas/blob/master/docs/running_grafeas.md).

File: _docs/guides/create-notes-occurrences.md [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+    return apiInstance.createNote(parent, body);
+  }
+```
+
+To verify that you successfully created your note, you can list all of the notes
+in a given project:
+
+```java
+/**
```
Comment (vtsao) [inline _docs/guides/create-notes-occurrences.md]: fix alignment

File: _docs/guides/create-notes-occurrences.md [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+    body.setNoteName(noteName);
+
+    return apiInstance.createOccurrence(parent, body);
+  }
+```
+
+You can list occurrences in a similar manner:
+
+```java
+/**
```
Comment (vtsao) [inline _docs/guides/create-notes-occurrences.md:237]: fix

File: _docs/guides/create-notes-occurrences.md [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+type: markdown
+---
+
+Before you can create [notes](/_docs/concepts/what-is-grafeas/overview.md) or
+[occurrences](/_docs/concepts/what-is-grafeas/overview.md), you'll need a
+running Grafeas server, which you can set up following
+[these instructions](https://github.com/grafeas/grafeas/blob/master/docs/running_grafeas.md).
+The examples below assume that you have a local server running on port 8080.
+
+## Create a project
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: ## Creating a project

File: _docs/guides/create-notes-occurrences.md [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+
+All notes and occurrences must be stored in a project. Generally, you should
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: s/must be/are

Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: Create separate projects for notes and occurrences so that setting up fine-grained access control is easier.

File: _docs/guides/create-notes-occurrences.md [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+create separate projects for notes and occurrences in order to make it easier to
+set up more fine-grained access control.
+
+To create a `provider_example` project to store notes in, run the following
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: **To create a project named `provider_example` to store the notes:**

File: _docs/guides/create-notes-occurrences.md [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+command:
+
+```bash
+curl -X POST http://localhost:8080/v1alpha1/projects \
+     -H "Content-Type: application/json" \
+     --data '{"name":"projects/provider_example"}'
+```
+
+Next, create an `occurrence_example` project:
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: **To create a project named `occurrence_example` to store the occurrences:**

File: _docs/guides/create-notes-occurrences.md [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+
+```bash
+curl -X POST http://localhost:8080/v1alpha1/projects \
+     -H "Content-Type: application/json" \
+     --data '{"name":"projects/occurrence_example"}'
+```
+
+## Using curl
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: How about the following format:

## Creating a note

### Using curl

### Using Java

## Creating an occurrence

### Using curl

### Using Java

File: _docs/guides/create-notes-occurrences.md [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+   * @param parent The project containing the note, of the format
+   *        "projects/<project_id>".
+   * @param name A user-specified identifier for the note
+   * @return a Note object representing the new note
+   * @throws Exception on errors while closing the client
+   */
+  public static ApiNote createNote(String parent, String name)
+         throws ApiException {
+    ApiNote body = new ApiNote();
+    body.setName(String.format("%s/notes/%s", parent, name));
```
Comment (vtsao) [inline _docs/guides/create-notes-occurrences.md]: Can you double check this, note creation request should only take in a note ID. Note name should be output only and ignored.

Also, kind is output only and shouldn't need to be set either.

File: _docs/guides/create-notes-occurrences.md [hunk 12]
```diff
@@ -0,0 +1,251 @@
...
+
+```java
+/**
+   * Creates and returns a new note
+   *
+   * @param parent The project containing the note, of the format
+   *        "projects/<project_id>".
+   * @param name A user-specified identifier for the note
+   * @return a Note object representing the new note
+   * @throws Exception on errors while closing the client
```
Comment (vtsao) [inline _docs/guides/create-notes-occurrences.md]: ApiException

File: _docs/guides/create-notes-occurrences.md [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+
+To verify that you successfully created your note, you can list all of the notes
+in a given project:
+
+```java
+/**
+   * Prints the notes in the specified project.
+   *
+   * @param parent the project to fetch notes from, of the form "projects/<project_id>".
+   * @throws Exception on errors while closing the client
```
Comment (vtsao) [inline _docs/guides/create-notes-occurrences.md]: ApiException and everywhere else below.

File: _docs/guides/create-notes-occurrences.md [hunk 14]
```diff
@@ -0,0 +1,251 @@
...
+in a valid note name for an existing note.
+
+```java
+/**
+  * Creates and returns a new occurrence.
+  *
+  * @param resourceUrl the Container Registry URL associated with the image
+  *                 example: "https://gcr.io/project/image@sha256:foo"
+  * @param parent the project to store this occurrence in, of the form
+  * "projects/<project_id>".
```
Comment (vtsao) [inline _docs/guides/create-notes-occurrences.md]: Fix the indent on these.

File: _docs/guides/create-notes-occurrences.md [hunk 15]
```diff
@@ -0,0 +1,251 @@
...
+### Create an occurrence
+
+You can create an occurrence with the following code. Note that you need to pass
+in a valid note name for an existing note.
+
+```java
+/**
+  * Creates and returns a new occurrence.
+  *
+  * @param resourceUrl the Container Registry URL associated with the image
```
Comment (vtsao) [inline _docs/guides/create-notes-occurrences.md]: Remove "Container Registry", just say a resource URL like a Docker image, example:...

File: _docs/guides/create-notes-occurrences.md [hunk 16]
```diff
@@ -0,0 +1,251 @@
...
+  *                 example: "https://gcr.io/project/image@sha256:foo"
+  * @param parent the project to store this occurrence in, of the form
+  * "projects/<project_id>".
+  * @param noteName the full name of the note this occurrence is attached to, of
+  * the form "projects/<project_id>/notes/<note_id>"
+  * @return an ApiOccurrence object representing the new occurrence
+  * @throws Exception on errors while closing the client
```
Comment (vtsao) [inline _docs/guides/create-notes-occurrences.md]: This should probably say something like "ApiException if there were errors with the API request."

**Review by vtsao — CHANGES_REQUESTED** [review]

File: _docs/guides/create-notes-occurrences.md (hunk 1 above)
Comment (avh121) [reply _docs/guides/create-notes-occurrences.md:5]: "Order: 15"? I'm not actually sure, but included it because it was included in the other docs markdown files.

@vtsao do you know?

File: _docs/guides/create-notes-occurrences.md (hunk 10 above)
Comment (avh121) [reply _docs/guides/create-notes-occurrences.md]: That seems reasonable to me; done. I couldn't decide which header should be at which level :)

File: _docs/guides/create-notes-occurrences.md (hunk 11 above)
Comment (avh121) [reply _docs/guides/create-notes-occurrences.md]: Okay. What fields does it make sense to set on a new note, then? I was just setting name and kind as examples. I also only see a "name" field, not an "id" in the [ApiNote](https://github.com/grafeas/client-java/blob/473ca607e3b55effccc046facba1b7492885c335/src/main/java/io/grafeas/model/ApiNote.java) class. Am I missing something?

File: _docs/guides/create-notes-occurrences.md (hunk 1 above)
Comment (vtsao) [reply _docs/guides/create-notes-occurrences.md:5]: I don't know, I actually have no idea what any of this metadata is for or if it's needed. Maybe you can find if it talks about it here: https://help.github.com/categories/github-pages-basics/

File: _docs/guides/create-notes-occurrences.md (hunk 11 above)
Comment (vtsao) [reply _docs/guides/create-notes-occurrences.md]: That's strange, the fields in the create note request object aren't anywhere in the generated client. This is probably some issue with our Swagger API ref and client generation that we can look at later. Let's just leave this code snippet as is for now.

**Review by vtsao — APPROVED** [review]

**Comment by vtsao** [comment]
Nitya, PTAL before I merge.

File: _docs/guides/create-notes-occurrences.md (hunk 1 above)
Comment (avh121) [reply _docs/guides/create-notes-occurrences.md:5]: Hm, it seems to have something to do with how we set up our navigation bar - we must be iterating over the pages at some point in order to list them in the nav bar. When I remove it, I see errors asking me to set the order. [This stack overflow post](https://stackoverflow.com/questions/13266369/how-to-change-the-default-order-pages-in-jekyll) seems relevant. 

tl;dr: leaving it in for now, the "15" is arbitrary and it seems we just need some number there, but something to keep in mind if and when we refactor the Grafeas site at some point.

File: _docs/guides/create-notes-occurrences.md [hunk 17]
```diff
@@ -0,0 +1,252 @@
...
+title: How to create notes and occurrences
+overview: Describes how to create notes and occurrences once you have a running Grafeas server.
+
+order: 15
+
+layout: docs
+type: markdown
+---
+
+This guide walks you through creating
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: Add a paragraph about why the user would need to create notes and occurrences.

File: _docs/guides/create-notes-occurrences.md [hunk 18]
```diff
@@ -0,0 +1,252 @@
...
+that your note was created, you can list all notes in a given project with the
+command:
+
+```bash
+curl http://localhost:8080/v1alpha1/projects/provider_example/notes
+```
+
+### Using Java
+
+The code sample below uses Java, but the Go, Python, and Ruby client libraries
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: s/sample/example

File: _docs/guides/create-notes-occurrences.md [hunk 19]
```diff
@@ -0,0 +1,252 @@
...
+work in a similar way. You can add the following methods to your
+[GrafeasApiExample](https://github.com/nhayes/client-java/#getting-started)
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: this is the first time you're mentioning GrafeasApiExample. can we introduce it by giving more context on what the example does?

File: _docs/guides/create-notes-occurrences.md [hunk 20]
```diff
@@ -0,0 +1,252 @@
...
+class, or the equivalent class that you're using to talk to the client library.
+
+When calling the below method, you'll need to specify the `provider_example`
+project with its full name: "projects/provider_example".
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: `projects/provider_example`

File: _docs/guides/create-notes-occurrences.md (hunk 19 above)
Comment (avh121) [reply _docs/guides/create-notes-occurrences.md]: Done, let me know what you think. 

I didn't include much context because it's a class that has a very simple 'add a note' method that is part of the installation instructions for the client-java repo -- so if you're using the client-java library, you'll need to have gone through that already.

File: _docs/guides/create-notes-occurrences.md (hunk 17 above)
Comment (avh121) [reply _docs/guides/create-notes-occurrences.md]: Done.

File: _docs/guides/create-notes-occurrences.md [hunk 21]
```diff
@@ -0,0 +1,259 @@
...
+order: 15
+
+layout: docs
+type: markdown
+---
+
+Notes and occurrences are the basic building blocks of storing data in Grafeas.
+Creating and managing them allows you to track information about your project.
+A _[note](/_docs/concepts/what-is-grafeas/overview.md)_ describes high-level
+information, whereas an _[occurrence](/_docs/concepts/what-is-grafeas/overview.md)_
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: perhaps be more specific here:

A note describes the high-level information about a metadata and an occurrence describes the occurrence of the note in the project.

File: _docs/guides/create-notes-occurrences.md [hunk 22]
```diff
@@ -0,0 +1,259 @@
...
+title: How to create notes and occurrences
+overview: Describes how to create notes and occurrences once you have a running Grafeas server.
+
+order: 15
+
+layout: docs
+type: markdown
+---
+
+Notes and occurrences are the basic building blocks of storing data in Grafeas.
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: s/data/metadata?

File: _docs/guides/create-notes-occurrences.md [hunk 23]
```diff
@@ -0,0 +1,260 @@
...
+overview: Describes how to create notes and occurrences once you have a running Grafeas server.
+
+order: 15
+
+layout: docs
+type: markdown
+---
+
+Notes and occurrences are the basic building blocks of storing metadata in
+Grafeas. Creating and managing them allows you to track information about your
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: Creating and managing them...: i think that line is a little distracting. should we delete it?

File: _docs/guides/create-notes-occurrences.md [hunk 24]
```diff
@@ -0,0 +1,260 @@
...
+project. A _[note](/_docs/concepts/what-is-grafeas/overview.md)_ the high-level
```
Comment (nityaravi) [inline _docs/guides/create-notes-occurrences.md]: A note represents

**Review by nityaravi — APPROVED** [review]

File: _docs/guides/create-notes-occurrences.md (hunk 23 above)
Comment (avh121) [reply _docs/guides/create-notes-occurrences.md]: Sure.
